### PR TITLE
Unstructured File Loader Support - USF

### LINKED ISSUE #808: UnstructuredFileLoader to Support .txt Files
### 🚀 The feature

In order to enhance the flexibility and usability of EmbedChain, it would be beneficial to introduce an UnstructuredFileLoader that can handle .txt files. This feature would allow users to easily import and work with unstructured text data within the EmbedChain environment.

### Motivation, pitch

Benefits:
Ease of Use: Users with data stored in .txt files can directly load their data without the need for conversion to a supported format.
Flexibility: This feature would cater to a broader range of data types and sources, making EmbedChain more adaptable to different use cases.
Consistency: By providing a dedicated loader for unstructured text files, we ensure a consistent approach to data loading and handling within EmbedChain.

### PR DESCRIPTION
**Pull request by MUZAMMILPERVAIZ**
## Description

I have added UnstructuredFileLoader for handling unstrcutured content. datatype: USF

Fixes # (issue)

## Type of change

- [* ] New feature

## How Has This Been Tested?

I have tested it in my APP.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #808 
- [ ] Made sure Checks passed

### CODE REVIEW

**Comment by deshraj** [comment]
Thanks @MUZAMMILPERVAIZ for the PR. Can you please add tests for this PR?

**Comment by MUZAMMILPERVAIZ** [comment]
TEST:

image

File: README.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ Embedchain empowers you to create ChatGPT like apps, on your own dynamic dataset
 * Doc file
 * Code documentation website loader
 * Notion and many more.
+* Unstructured
```
Comment (deshraj) [inline README.md:51]:
```suggestion
* Unstructured file loader and many more
```

File: README.md [hunk 2]
```diff
@@ -47,6 +47,7 @@ Embedchain empowers you to create ChatGPT like apps, on your own dynamic dataset
 * Doc file
 * Code documentation website loader
 * Notion and many more.
```
Comment (deshraj) [inline README.md:50]:
```suggestion
* Notion
```

File: embedchain/models/data_type.py [hunk 3]
```diff
@@ -25,6 +25,7 @@ class IndirectDataType(Enum):
     CSV = "csv"
     MDX = "mdx"
     IMAGES = "images"
+    USF = 'USF'
```
Comment (deshraj) [inline embedchain/models/data_type.py]:
```suggestion
    USF = 'unstructured'
```

**Review by deshraj — APPROVED** [review]
Looks good. Thanks!

File: embedchain/data_formatter/data_formatter.py [hunk 4]
```diff
@@ -77,6 +79,7 @@ def _get_loader(self, data_type: DataType, config: LoaderConfig) -> BaseLoader:
             DataType.CSV: CsvLoader,
             DataType.MDX: MdxLoader,
             DataType.IMAGES: ImagesLoader,
+            DataType.UNSTRUCTURED: UnstructuredLoader
```
Comment (deshraj) [inline embedchain/data_formatter/data_formatter.py:82]:
```suggestion
            DataType.UNSTRUCTURED: UnstructuredLoader,
```

File: embedchain/data_formatter/data_formatter.py [hunk 5]
```diff
@@ -119,6 +122,7 @@ def _get_chunker(self, data_type: DataType, config: ChunkerConfig) -> BaseChunke
             DataType.MDX: MdxChunker,
             DataType.IMAGES: ImagesChunker,
             DataType.XML: XmlChunker,
+            DataType.UNSTRUCTURED: UnstructuredFileChunker
```
Comment (deshraj) [inline embedchain/data_formatter/data_formatter.py:125]:
```suggestion
            DataType.UNSTRUCTURED: UnstructuredFileChunker,
```

**Comment by deshraj** [comment]
Seems like there is an issue with the build failing. I will fix it as part of my follow up PR.

**Pull request merged**